### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Roman-Malysh**
https://roman-malysh.github.io/react_static-list-of-todos/

### CODE REVIEW

File: src/components/UserInfo/UserInfo.tsx [hunk 1]
```diff
@@ -1,4 +1,13 @@
-// Add the required types and props
-export const UserInfo = () => (
-  <>UserInfo markup</>
+import React from 'react';
+import { User } from '../../types/User';
+
+interface Props {
+  user: User | null;
```
Comment (TarasHoliuk) [inline src/components/UserInfo/UserInfo.tsx:5]:
```suggestion
  user: User;
```
check the user in the `TodoInfo` component instead and don't render `UserInfo` component at all if the user doesn't exist

File: src/components/UserInfo/UserInfo.tsx [hunk 2]
```diff
@@ -1,4 +1,13 @@
...
+}
+export const UserInfo : React.FC<Props>= ({user}) => (
+  user && (
+    <a className="UserInfo" href={`mailto:${user.email}`}>
```
Comment (TarasHoliuk) [inline src/components/UserInfo/UserInfo.tsx:9]: you may use destructuring assignment here

File: src/components/TodoList/TodoList.tsx [hunk 3]
```diff
@@ -1,4 +1,16 @@
...
-  <>TodoList markup</>
+import React from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoInfo } from '../TodoInfo';
+
+type Props = {
+  todos: Todo[];
+}
+export const TodoList: React.FC<Props> = ({ todos }) => (
+  <section className="TodoList">
```
Comment (TarasHoliuk) [inline src/components/TodoList/TodoList.tsx:9]: when you mapping (render a list) use semantic tags (`ul`, `li`)

File: src/components/TodoInfo/TodoInfo.tsx [hunk 4]
```diff
@@ -1,4 +1,24 @@
...
+import { UserInfo } from '../UserInfo';
+
+interface Props {
+  todo: Todo;
+}
+export const TodoInfo:React.FC <Props> = ({ todo }) => {
+  const { title, completed, user } = todo;
+
+  return (
+    <article className={classNames(
```
Comment (TarasHoliuk) [inline src/components/TodoInfo/TodoInfo.tsx:13]: when you mapping (render a list) use semantic tags (`ul`, `li`)

**Review by TarasHoliuk — CHANGES_REQUESTED** [review]

File: src/components/TodoList/TodoList.tsx [hunk 5]
```diff
@@ -1,4 +1,17 @@
...
+
+type Props = {
+  todos: Todo[];
+};
+
+export const TodoList: React.FC<Props> = ({ todos }) => (
+  <section className="TodoList">
+    <ul>
+      {todos.map((todo) => {
+        return <li><TodoInfo key={todo.id} todo={todo} /></li>;
```
Comment (tetiana-ivanchenko) [inline src/components/TodoList/TodoList.tsx:13]:
```suggestion
        return <li key={todo.id}><TodoInfo todo={todo} /></li>;
```

File: src/components/UserInfo/UserInfo.tsx [hunk 6]
```diff
@@ -1,4 +1,11 @@
...
+import React from 'react';
+import { User } from '../../types/User';
+
+interface Props {
+  user: User;
+}
+export const UserInfo : React.FC<Props> = ({ user }) => (
+  <a className="UserInfo" href={`mailto:${user.email}`}>
+      {user.name}
+    </a>
```
Comment (tetiana-ivanchenko) [inline src/components/UserInfo/UserInfo.tsx:10]:
```suggestion
  <a className="UserInfo" href={`mailto:${user.email}`}>
    {user.name}
  </a>
```

File: src/components/TodoInfo/TodoInfo.tsx [hunk 7]
```diff
@@ -1,4 +1,25 @@
...
+      'TodoInfo',
+      {
+        'TodoInfo--completed': completed,
+      },
+    )}
+    >
+      <h2 className="TodoInfo__title">{title}</h2>
+      {user &&
+      <UserInfo user={user} />}
```
Comment (tetiana-ivanchenko) [inline src/components/TodoInfo/TodoInfo.tsx:22]:
```suggestion
      {user && <UserInfo user={user} />}
```

**Review by tetiana-ivanchenko — CHANGES_REQUESTED** [review]

File: src/components/TodoList/TodoList.tsx (hunk 5 above)
Comment (Roman-Malysh) [reply src/components/TodoList/TodoList.tsx:13]: what will be the difference?

Comment (sTorba24) [reply src/components/TodoList/TodoList.tsx]: we are setting key attibute for tag `<li> `

File: src/components/TodoList/TodoList.tsx [hunk 8]
```diff
@@ -1,4 +1,17 @@
...
+
+type Props = {
+  todos: Todo[];
+};
+
+export const TodoList: React.FC<Props> = ({ todos }) => (
+  <section className="TodoList">
+    <ul>
+      {todos.map((todo) => {
+        return <li key={todo.id}><TodoInfo todo={todo} /></li>;
```
Comment (sTorba24) [inline src/components/TodoList/TodoList.tsx:13]: use just todoInfo and move `<li>` to TodoInfo marckup

**Review by sTorba24 — CHANGES_REQUESTED** [review]
Fix previous comments please

File: src/components/TodoList/TodoList.tsx (hunk 8 above)
Comment (Roman-Malysh) [reply src/components/TodoList/TodoList.tsx:13]: why u guys suggesting different things which have no impact on global picture, I did exactly what ur colleague asked above. I more than sure that it is fine

Comment (TarasHoliuk) [reply src/components/TodoList/TodoList.tsx:13]: if you mean this https://github.com/mate-academy/react_static-list-of-todos/pull/1617#discussion_r1126025213, that was about `key` property. Tanya suggested moving the `key` to the first (highest) tag/component inside `map`'s callback.

Otherwise, the `key` property doesn't make any sense

But that doesn't negate that structure of tags should be a bit different. It doesn't make an impact on the global result. But that's how clean, maintainable, and readable code should be written.

Summing up: you had to use `ul` instead of `section`, not inside, i.e `<ul className="TodoList">`. `li` should be inside the `TodoInfo`, instead of the `article` tag.

I'll approve this PR now, but please, take this comment into account in the future.

**Review by TarasHoliuk — APPROVED** [review]
